src/database/DataTypes/Requisition.js

```diff
@@ -39,11 +42,14 @@ export class Requisition extends Realm.Object {
   }
 
   /**
-   * Delete requisition and associated requisition items.
+   * Delete requisition and associated requisition items/period.
    *
    * @param {Realm} database
    */
   destructor(database) {
+    if (this.period) {
```

Comment:
once again, no need to removeRequisition from period

Reply:
done 62f79d382403c5df4850c98e87ff0a034b1f7fde